app/src/main/java/at/roteskreuz/stopcorona/screens/questionnaire/QuestionnaireViewModel.kt

```diff
@@ -106,9 +107,13 @@ class QuestionnaireViewModel(
 
     fun observeDecision(): Observable<Decision> = executeDecisionSubject
 
-    fun observeFetchState(): Observable<State> = questionnaireStateObserver.observe()
+    fun observeFetchState(): Observable<State> = fetchQuestionnaireStateObserver.observe()
 
-    fun observeQuestionnaireWithAnswers(language: ConfigurationLanguage): Observable<List<DbQuestionnaireWithAnswers>> {
-        return configurationRepository.observeQuestionnaireWithAnswers(language)
+    fun observeQuestionnaireWithQuestions(): Observable<List<DbQuestionnaireWithAnswers>> = questionnaireSubject
+
+    fun getQuestionnaireWithAnswers(language: ConfigurationLanguage) {
+        launch {
+            questionnaireSubject.onNext(configurationRepository.getQuestionnaireWithAnswers(language))
```

Comment:
This call could trigger an exception that is not handled. We can move the logic of getting the questionnaire with answers inside the `fetchQuestionnaire` method

Reply:
done